src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,93 @@
+<!--
+***********************************************************************************************
+Microsoft.Android.Sdk.Bindings.Gradle.targets
+
+This file contains MSBuild targets that support building and operating on Android Gradle projects.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.Gradle"              AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <_BuildAndroidGradleProjectsStamp Condition=" '$(_BuildAndroidGradleProjectsStamp)' == '' ">$(IntermediateOutputPath)stamp/_BuildAndroidGradleProjects.stamp</_BuildAndroidGradleProjectsStamp>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <AndroidGradleProjectReference>
+      <Bind>true</Bind>
+      <Configuration>Release</Configuration>
+      <ReferenceLibraryOutputs>true</ReferenceLibraryOutputs>
+      <ModuleName></ModuleName>
+      <OutputPath>$(MSBuildProjectDirectory)/$(OutputPath)gradle/</OutputPath>
+      <Pack>true</Pack>
+      <Visible>false</Visible>
+    </AndroidGradleProjectReference>
+  </ItemDefinitionGroup>
+
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs">
+    <ItemGroup>
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
```

Comment:
Should one of the inputs be all the `@(AndroidGradleProjectReference)` items? You could write them out to a file in `obj` with `<WriteLinesToFile WriteOnlyWhenDifferent="true")/>`

Reply:
Would this be done to ensure a rebuild happens if new items are added or an include value changes? Would this also account for metadata items changing? I was hoping to handle this type of change by adding the project file to the input list, though I think this wouldn't handle items defined in a separate targets file. Is there a property or item that can be used to get the project file and any such imports?

Reply:
You could list `@(_AndroidMSBuildAllProjects)`, and that would handle if any MSBuild file changes. That might be fine.

Reply:
Updated this to include $(MSBuildProjectFullPath) and @(_AndroidMSBuildAllProjects)